### PR TITLE
updated index_list2 function logic to cater index configs

### PR DESCRIPTION
**Pull request by wanlok**

### CODE REVIEW

File: functions/src/list_index.ts [hunk 1]
```diff
@@ -62,71 +62,81 @@ export const index_list = functions.runWith({timeoutSeconds:540}).https
...
     const listId:string=request.query.list as string;
-    const fieldId:string=request.query.field as string;
-    const settings:any=request.body;
-
-    console.log(`indexing list ${listId} by ${fieldId} with settings ${JSON.stringify(settings)}`)
-    const items=
-    await db.collection("list").doc(listId).collection('item').get();
-
+    let indices = await db.collection('index').where('listId', '==', listId).get();
+    indices.forEach(index => {
```
Comment (kovarny) [inline functions/src/list_index.ts:67]: The problem with this code is that it is async and will start running in parallel while the rest of the code will be running at the same time. We have to first delete the docs and only then proceed to creating the new index.

**Review by kovarny — CHANGES_REQUESTED** [review]

File: functions/src/list_index.ts (hunk 1 above)
Comment (wanlok) [reply functions/src/list_index.ts]: I have followed your sample code to use batch delete. Please review.

File: functions/src/list_index.ts [hunk 2]
```diff
@@ -62,71 +63,75 @@ export const index_list = functions.runWith({timeoutSeconds:540}).https
...
-
     const listId:string=request.query.list as string;
-    const fieldId:string=request.query.field as string;
-    const settings:any=request.body;
-
-    console.log(`indexing list ${listId} by ${fieldId} with settings ${JSON.stringify(settings)}`)
-    const items=
-    await db.collection("list").doc(listId).collection('item').get();
-
+    deleteLargeColByQuery(db.collection('index').where('listId', '==', listId));
```
Comment (kovarny) [inline functions/src/list_index.ts]: does this need await here?

**Review by kovarny — APPROVED** [review]
Looks good.

**Pull request merged**